### PR TITLE
Add type=auto using client Accept header to auto negotiate type.

### PR DESCRIPTION
**Pull request by barnybug**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -485,7 +485,7 @@ Image measures are always in pixels, unless otherwise indicated.
 - **text**        `string` - Watermark text content. Example: `copyright (c) 2189`
 - **font**        `string` - Watermark text font type and format. Example: `sans bold 12`
 - **color**       `string` - Watermark text RGB decimal base color. Example: `255,200,150`
-- **type**        `string` - Specify the image format to output. Possible values are: `jpeg`, `png` and `webp`
+- **type**        `string` - Specify the image format to output. Possible values are: `jpeg`, `png`, `webp` and `auto`. `auto` will use the preferred format requested by the client in the HTTP Accept header.
```
Comment (h2non) [inline README.md]: Can we specify here that multiple values are allowed comma-separated?

File: controllers.go [hunk 2]
```diff
@@ -52,6 +52,26 @@ func imageController(o ServerOptions, operation Operation) func(http.ResponseWri
 	}
 }
 
+func determineAcceptMimeType(accept string) string {
+	for _, mime := range strings.Split(accept, ",") {
+		// strip off any other parameters after mime type
+		i := strings.Index(mime, ";")
```
Comment (h2non) [inline controllers.go]: Consider using: https://golang.org/pkg/mime/#ParseMediaType

**Comment by h2non** [comment]
I like this. Thanks for the work here. Just consider my comments and we are ready to go.

File: controllers.go (hunk 2 above)
Comment (barnybug) [reply controllers.go]: Neat, didn't know about that. Thanks!

**Comment by h2non** [comment]
Thank you! This feature will be available in the upcoming release.

**Comment by h2non** [comment]
Feature shipped in imaginary `v1.0.11`. Thanks!

**Comment by barnybug** [comment]
Nice, many thanks to you for imaginary!

**Pull request merged**